### DISCUSSION TITLE
`adapter-svelte` with SvelteKit: Using TypeScript and `app.d.ts`

### DISCUSSION THREAD

**Discussion by oscard0m** [ivanhofer/typesafe-i18n#717] [Q&A] [answered]
When integrating `typesafe-i18n` with SvelteKit I was facing a TypeScript problem:
`locale` was resolved as `string` instead of `Locales`

After checking existing [SvelteKit example with `typesafe-i18n`](https://github.com/ivanhofer/typesafe-i18n-demo-sveltekit) I noticed the issue was I was defining a wrong `app.d.ts`:

```diff
// app.d.ts
+ type Locales = import('$i18n/i18n-types').Locales;
+ type TranslationFunctions = import('$i18n/i18n-types').TranslationFun;
  declare global {
     namespace App {
        interface Locals {
-          locale: string;
+          locale: Locales;
+          LL: TranslationFunctions;
        }
     }
  }
```

Would it make sense to auto-generate this file? Is it even possible? Why is this `app.d.ts` necessary for SvelteKit but not other frameworks?

**Comment by ivanhofer** [comment]
> Would it make sense to auto-generate this file? Is it even possible? 

Auto-generating the file is not an option as it would override existing modifications to that file.
We could add a section to the docs: https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/adapter-svelte
Do you want to add that?

> Why is this app.d.ts necessary for SvelteKit but not other frameworks?

`SvelteKit` cares about `TypeScript` and provides options to strongly type almost everything. Other frameworks are more relaxed.
Depending on how you set `typesafe-i18n` up for other frameworks, you may also need to add some type declarations.

**Reply by oscard0m** [reply, marked as answer]
👍🏽 Done in:
- https://github.com/ivanhofer/typesafe-i18n-demo-sveltekit/pull/39
- https://github.com/ivanhofer/typesafe-i18n/pull/724
